### DISCUSSION TITLE
Add ability to update atom outside component

### DISCUSSION THREAD

**Discussion by Ma5ha** [pmndrs/jotai#2744] [Ideas]
Currently only way to update atom value is to identify it in a hook and then use setter function.

Like this:

```
export const COUNTER_JOTAI = atom<number>(0);
const [counter, setCounter] = useAtom(COUNTER_JOTAI);
  function handleAddCounter(number: number) {
    setCounter(++number);
  }

```

I think adding ability to update it outside like  React Hook State  would be beneficial. It would give us easier interface and it would play better with some other libs. 

It could look something like this:   

```
export const COUNTER_JOTAI = atom<number>(0);  
 COUNTER_JOTAI.set(6) //or
 COUNTER_JOTAI.set( COUNTER_JOTAI.get() +1)
```

**Comment by rothsandro** [comment]
You can already work with atoms outside of components by accessing the store: https://jotai.org/docs/core/store

**Reply by Ma5ha** [reply]
I do know that, but interface of that is not 'natural' so to say. It feels like hack.

**Reply by rothsandro** [reply]
What part of it does feel like a hack?

The `get()` / `set()` functions you propose would have to use a store to get/set the value, because the atom itself doesn't hold the value. With an additional param for the store, the API wouldn't be much different:

```js
store.set(myAtom, 42);

// vs.

myAtom.set(24, store);
```

**Reply by dmaskasky** [reply]
```ts
import { atom, getDefaultStore } from 'jotai'

export function atomWithAccessors(
  readOrValue,
  writeOrUndefined,
  store = getDefaultStore()) {
  const baseAtom = atom(...[
    readOrValue,
    writeOrUndefined,
  ].filter(Boolean))
  return Object.assign(
    baseAtom,
    {
      get() {
        return store.get(baseAtom)
      },
      set(...values) {
        return store.set(baseAtom, ...values)
      },
    }
  )
}
```

**Reply by Ma5ha** [reply]
@rothsandro  I do think what you are saying is sensible. This is not hard to create on my own.

 The hack part is mostly related when you are getting a default store.  So` get()` / ` set()` would not be using second parameter for that.  For default store it would be 

```javascript
myAtom.set(33)
```

For custom store we could pass second argument. 

```javascript
myAtom.set(33, customStore)
```

To me this looks like a helpful interface to be shipped  with lib. As this is something that I create on my own usually. Or it could be utility function as @dmaskasky suggested.
